x-pack/plugins/alerting/public/pages/maintenance_windows/helpers/get_weekday_info.test.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import moment from 'moment';
+import { getWeekdayInfo } from './get_weekday_info';
+
+describe('getWeekdayInfo', () => {
+  test('should return the fourth tuesday of the month for 11/23/2021', () => {
+    expect(getWeekdayInfo(moment('2021-11-23'))).toEqual({
+      dayOfWeek: 'Tuesday',
+      isLastOfMonth: false,
+      nthWeekdayOfMonth: 4,
+    });
+  });
+
+  test('should return the third Tuesday of the month 11/16/2021', () => {
+    expect(getWeekdayInfo(moment('2021-11-16'))).toEqual({
+      dayOfWeek: 'Tuesday',
+      isLastOfMonth: false,
+      nthWeekdayOfMonth: 3,
+    });
+  });
+
+  test('should return the last Friday of the month 12/25/2020', () => {
+    expect(getWeekdayInfo(moment('2020-12-25'))).toEqual({
+      dayOfWeek: 'Friday',
+      isLastOfMonth: true,
+      nthWeekdayOfMonth: 4,
+    });
+  });
+
+  test('should return expected invalid props for a null date', () => {
+    expect(getWeekdayInfo(moment(null))).toEqual({
```

Comment:
No I don't think it's possible to get a null date, because it comes from the start date. This function is just used to set weekday labels for the form, so I am not sure if we want to throw an error here